osprey/search_space.py

```diff
@@ -30,17 +30,17 @@ def __init__(self):
     def n_dims(self):
         return len(self.variables)
 
-    def add_jump(self, name, min, max, step):
+    def add_jump(self, name, min, max, step, var_type=float):
         """ An integer/float valued dimension bounded between
         range(min,max,step). Note that the right endpoint of the interval
-        includes `max`. This is a wrapper around the add_enum.
+        includes `max`. This is a wrapper around the add_enum. It assumes
+        that the jump is float but can also use ints. 
         """
-        try:
-            min, max, step = map(float, (min, max, step))
-        except:
-            raise ValueError('Variable %s must be either an int or a float'
-                             % name)
-        choices = np.arange(min, max + 1, step)
+        if var_type != float and var_type != int:
```

Comment:
@mpharrigan is right that this is probably not even necessary, since numpy will check for numeric types anyways